### ISSUE TITLE
Add option for object-literal-sort-keys to not ignore blank lines

### ISSUE THREAD

**Issue by dmitryshimkin** [palantir/tslint#4756]
### Bug Report

- __TSLint version__: 5.17.0
- __TypeScript version__: 2.7.2
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
const test = {
  b: 'b',

  a: 'a',

  c: 'c',
};
```

with `tslint.json` configuration:

```json
"object-literal-sort-keys": [
  true,
  "ignore-case",
  "locale-compare"
]
```

#### Actual behavior

No lint errors

#### Expected behavior

"Properties are not sorted" error.

**Comment by localnet** [comment]
After see the [rule documentation](https://palantir.github.io/tslint/rules/object-literal-sort-keys/) seems the correct behavior:

> When using the default alphabetical ordering, additional blank lines may be used to group object properties together while keeping the elements within each group in alphabetical order.

**Comment by dmitryshimkin** [comment]
> After see the [rule documentation](https://palantir.github.io/tslint/rules/object-literal-sort-keys/) seems the correct behavior:
> 
> > When using the default alphabetical ordering, additional blank lines may be used to group object properties together while keeping the elements within each group in alphabetical order.

Right, according to the docs it seems to be expected. Then I wonder if there is a way to opt out this. Otherwise this rule cannot be used in cases when there is no any groups, but _all_ properties are separated with blank lines for better readability.

**Comment by JoshuaKGoldberg** [comment]
Seems like a reasonable idea - accepting PRs to add an option to _not_ ignore blank lines.

Do keep #4534 in mind: custom rules including forks of built-in rules such as `object-literal-sort-keys` are always an option and can be maintained outside this repo.
